### PR TITLE
refactor: divide pipeline related tests

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/tests/internal/e2etests/test_component.py [hunk 1]
```diff
@@ -99,3 +99,37 @@ def test_component_load(
...
+
+    def test_component_with_default_label(
+        self,
+        client: MLClient,
+        randstr: Callable[[str], str],
+        yaml_path: str,
+    ) -> None:
+        component_name = randstr("component_name")
+
+        create_component(client, component_name, path=yaml_path)
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/tests/internal/e2etests/test_component.py:111]: component = client.components.get(component_name)
node = component()


component = client(label="default")

File: sdk/ml/azure-ai-ml/tests/internal/e2etests/test_component.py [hunk 2]
```diff
@@ -99,3 +99,37 @@ def test_component_load(
...
+        assert default_component.version == "azureml_default"
+
+        # plan B: use a protected field to indicate the label
+        # pros:
+        # 1. can be used to update component
+        # 2. hide keyword from user
+        # 3. unified solution for all component label
+        # pros:
+        # 1. introduce a new "protected" field like "_is_anonymous"
+        assert default_component._label == "default"
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/tests/internal/e2etests/test_component.py:131]: remove this

File: sdk/ml/azure-ai-ml/tests/internal/e2etests/test_component.py [hunk 3]
```diff
@@ -99,3 +99,37 @@ def test_component_load(
...
+        # or add a special logic in job_operations so all behaviour is hidden from public interface?
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/tests/internal/e2etests/test_component.py:132]: no need to add special logic

File: sdk/ml/azure-ai-ml/tests/internal/e2etests/test_component.py (hunk 1 above)
Comment (elliotzh) [reply sdk/ml/azure-ai-ml/tests/internal/e2etests/test_component.py:111]: "latest"

"default"

File: sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/helloworld_pipeline_job_with_default_component.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  hello_world_component:
+    type: command
+    comment: arbitrary string
+    component: azureml:microsoftsamples_command_component_basic@default
+    compute: azureml:cpu-cluster
+    inputs:
+      component_in_number: ${{parent.inputs.job_in_number}}
+      component_in_path: ${{parent.inputs.job_in_path}}
+
+# should we support azureml:microsoftsamplesCommandComponentBasic_second?
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/helloworld_pipeline_job_with_default_component.yml:34]: @wangchao1230 please help confirm this

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_operation_orchestrator.py [hunk 5]
```diff
@@ -356,6 +372,9 @@ def _resolve_name_version_from_name_label(self, aml_id: str, azureml_type: str)
         ):
             return aml_id, None
 
+        if label == "default":
```
Comment (brynn-code) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_operation_orchestrator.py:375]: DEFAULT_LABEL_NAME?

File: sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/helloworld_pipeline_job_with_default_component.yml (hunk 4 above)
Comment (wangchao1230) [reply sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/helloworld_pipeline_job_with_default_component.yml:34]: we should not support this

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_operation_orchestrator.py (hunk 5 above)
Comment (brynn-code) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_operation_orchestrator.py:375]: Will this line affect other type of assets which also has default label, like Data?

**Review by brynn-code — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_operation_orchestrator.py (hunk 5 above)
Comment (elliotzh) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_operation_orchestrator.py:375]: sorry, this line belongs to an earlier implementation and should be removed.

Comment (elliotzh) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_operation_orchestrator.py:375]: the same as below

**Pull request merged**